### ISSUE TITLE
Micrometer configuration lifecycle violation

### ISSUE THREAD

**Issue by jonatan-ivanov** [eclipse-vertx/vert.x#5268]
Micrometer's `MeterRegistry` should be "configured" (see `MeterRegistry.Config`) before registering `Meter` instances to it. For example `MeterFilter`s should be configured before a Meter has been registered to the registry.

Historically, Micrometer has allowed configuring `MeterFilter`s after meters have been registered but due to certain performance optimizations with https://github.com/micrometer-metrics/micrometer/pull/4917 we took a step toward warning users about this situation.

See:
- https://github.com/micrometer-metrics/micrometer/issues/4920
- https://github.com/micrometer-metrics/micrometer/pull/4917

It seems in certain cases Vert.x configures a MeterFilter after a Meter has been registered to the registry causing the warning mentioned above. See https://github.com/micrometer-metrics/micrometer/issues/4920#issuecomment-2125854418 by @sanyarnd with a reproducer.

@sanyarnd could you please provide the Vert.x version you are using and any additional details you have?

**Comment by vietj** [comment]
@jonatan-ivanov can you re-create this issue in this repo please ? https://github.com/vert-x3/vertx-micrometer-metrics/issues/202

**Comment by jonatan-ivanov** [comment]
@vietj Of course: https://github.com/vert-x3/vertx-micrometer-metrics/issues/224
